### PR TITLE
Statistics on instantiations per quantified formula.

### PR DESCRIPTION
**Pull request by ajreynol**
This adds a new print format for instantiations --print-instantiations=num, which prints the total number of instantations of quantified formulas.  This count is user-context-dependent, which is in sync with the existing print-instantiation format (list).

It also simplifies and improves printing of Instantiation Tries.

### CODE REVIEW

File: src/theory/quantifiers/inst_match_trie.cpp [hunk 1]
```diff
@@ -392,7 +386,6 @@ bool CDInstMatchTrie::recordInstLemma(Node q,
 void CDInstMatchTrie::print(std::ostream& out,
                             Node q,
                             std::vector<TNode>& terms,
-                            bool& firstTime,
```
Comment (HanielB) [inline src/theory/quantifiers/inst_match_trie.cpp:395]: Comment on the PR description why you are removing this.

File: src/theory/quantifiers/instantiate.cpp [hunk 2]
```diff
@@ -264,9 +264,8 @@ bool Instantiate::addInstantiation(
     return false;
   }
 
-  d_total_inst_debug[q]++;
+  d_total_inst_debug[q] = d_total_inst_debug[q] + 1;
```
Comment (HanielB) [inline src/theory/quantifiers/instantiate.cpp:267]: Why this change?

File: src/theory/quantifiers/quantifiers_attributes.cpp [hunk 3]
```diff
@@ -50,8 +50,9 @@ void QuantAttributes::setUserAttribute( const std::string& attr, Node n, std::ve
     SygusAttribute ca;
     n.setAttribute( ca, true );
   }
-  else if (attr == "quant-name")
+  else if (attr == "qid")
   {
+    // using z3 syntax "qid"
```
Comment (HanielB) [inline src/theory/quantifiers/quantifiers_attributes.cpp:55]: I was going to ask why use this rather than `:named` but I think this comment answers the question. :)

File: src/options/quantifiers_options.toml [hunk 4]
```diff
@@ -1962,6 +1962,30 @@ header = "options/quantifiers_options.h"
...
+  type       = "PrintInstMode"
+  default    = "LIST"
+  help       = "print format for printing instantiations"
+  help_mode  = "Print format for printing instantiations."
+[[option.mode.LIST]]
+  name = "list"
+  help = "Print the list of instantiations per quantified formula."
+[[option.mode.NUM]]
+  name = "num"
+  help = "Print the total number of instantiations per quantified formula."
```
Comment (HanielB) [inline src/options/quantifiers_options.toml:1980]: Maybe helpful to say that only prints if more than 0 instantiations.

File: src/theory/quantifiers/instantiate.cpp (hunk 2 above)
Comment (ajreynol) [reply src/theory/quantifiers/instantiate.cpp:267]: because its context-dependent now, which doesnt support `++`.
